plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibraries.java

```diff
@@ -64,6 +78,53 @@ public static Library getLibrary(String id) {
     return libraries.get(id);
   }
 
+  private static List<Library> loadClientApis() {
+    Bundle bundle = FrameworkUtil.getBundle(CloudSdk.class);
+    URL url = bundle.getResource("/com/google/cloud/tools/libraries/libraries.json");
+    
+    try (InputStream in = url.openStream()) {
+      JsonReader reader = Json.createReader(in); 
+      JsonObject[] apis = reader.readArray().toArray(new JsonObject[0]); 
+      List<Library> clientApis = new ArrayList<>(apis.length);
+      for (JsonObject api : apis) {
+        String name = api.getString("name");
+        String id = api.getString("id");
+        Library library = new Library(id);
+        library.setGroup(CLIENT_APIS_GROUP);
+        library.setName(name);
+        JsonArray clients = api.getJsonArray("clients");
+        for (JsonObject client : clients.toArray(new JsonObject[0])) {
+          JsonString language = client.getJsonString("language");
+          if (language != null && "java".equals(language.getString())) {
+            String toolTip = client.getString("infotip");
+            library.setToolTip(toolTip);
+            JsonObject coordinates = client.getJsonObject("mavenCoordinates");
+            String groupId = coordinates.getString("groupId");
+            String artifactId = coordinates.getString("artifactId");
+            String versionString = coordinates.getString("version");
+
+            MavenCoordinates mavenCoordinates = new MavenCoordinates.Builder()
+                .setGroupId(groupId)
+                .setArtifactId(artifactId)
+                .setVersion(versionString)
+                .build();
+            LibraryFile file = new LibraryFile(mavenCoordinates);
+            List<LibraryFile> libraryFiles = new ArrayList<>();
+            libraryFiles.add(file);
+            library.setLibraryFiles(libraryFiles);
+            library.setResolved(false);
+            break;
+          }
+        }
+        clientApis.add(library);
+      }
+      
+      return clientApis;
+    } catch (IOException ex) {
```

Comment:
Should we catch `ClassCastException` in case the .json contains invalid data?

Reply:
What would throw a `ClassCastException` here? The `readArray` method might throw a `JSONException` though that's already a runtime exception; and would indicate a program bug since the JSON is loaded from the jar archive.

Reply:
[getString()](http://docs.oracle.com/javaee/7/api/javax/json/JsonObject.html#getString-java.lang.String-) is documented as throwing CCE if the object is not a string.